aspnetcore/blazor/hybrid/tutorials/maui-blazor-web-app.md

```diff
@@ -0,0 +1,392 @@
+
+## Using interfaces to support different device implementations
+
+The following example demonstrates how to use an interface to call into different implementations across the web app and the native (MAUI) app. The following example creates a component that displays the device form factor. Use the MAUI abstraction layer for native apps and provide an implementation for the web app.
+
+In the Razor class library (RCL), create an `Interfaces` folder and add file named `IFormFactor.cs` with the following code.
+
+`Interfaces/IFormFactor.cs`:
+
+:::code language="csharp" source="~/../blazor-samples/8.0/MauiBlazorWeb/MauiBlazorWeb.Shared/Interfaces/IFormFactor.cs":::
+
+In the RCL's `Components` folder, add the following `DeviceFormFactor` component.
+
+`Components/Pages/DeviceFormFactor.razor`:
+
+:::code language="razor" source="~/../blazor-samples/8.0/MauiBlazorWeb/MauiBlazorWeb.Shared/Components/Pages/DeviceFormFactor.razor":::
+
+In the RCL, add an entry for the `DeviceFormFactor` component to the navigation menu.
+
+In `Components/Layout/NavMenu.razor`:
+
+```razor
+<div class="nav-item px-3">
+    <NavLink class="nav-link" href="device-form-factor">
+        <span class="bi bi-list-nested-nav-menu" aria-hidden="true"></span> Form Factor
+    </NavLink>
+</div>
+```
+
+Provide implementations in the web and native apps.
+
+In the Blazor Web App, add a folder named `Services`. Add a file to the `Services` folder named `FormFactor.cs` with the following code.
+
+`Services/FormFactor.cs` (Blazor Web App project):
+
+:::code language="csharp" source="~/../blazor-samples/8.0/MauiBlazorWeb/MauiBlazorWeb.Web/Services/FormFactor.cs":::
+
+In the MAUI project, add a folder named `Services` and add a file named `FormFactor.cs`. The MAUI abstractions layer is used to write code that works on all native device platforms.
+
+`Services/FormFactor.cs` (MAUI project):
+ 
+:::code language="csharp" source="~/../blazor-samples/8.0/MauiBlazorWeb/MauiBlazorWeb.Maui/Services/FormFactor.cs":::
+
+Use dependency injection to obtain the implementations of these services.
+
+In the MAUI project, open the `MauiProgram.cs` file and add the following `using` statements to the top of the file:
+
+```csharp
+using MauiBlazorWeb.Maui.Services;
+using MauiBlazorWeb.Shared.Interfaces;
+```
+
+Immediately before the call to `builder.Build()`, add the following code to add device-specific services used by the RCL:
+
+```csharp
+builder.Services.AddSingleton<IFormFactor, FormFactor>();
+```
+
+In the Blazor Web App, open the `Program` file and add the following `using` statements to the top of the file:
+
+```csharp
+using MauiBlazorWeb.Shared.Interfaces;
+using MauiBlazorWeb.Web.Services;  
+```
+
+Immediately before the call to `builder.Build()`, add the following code to add device-specific services used by the RCL:
+
+```csharp
+builder.Services.AddScoped<IFormFactor, FormFactor>();
+```
+
```

Comment:
Should we also mention that if you are also targeting WebAssembly (you have a Web.Client project) you will also need to provide an implementation here as well?